### PR TITLE
Add Version v1.0.2

### PR DESCRIPTION
**Pull request by galal-hussein**
Cherry picks for rancher fixes to nginx v1.0.2

### CODE REVIEW

**Review by dereknola — APPROVED** [review]

**Review by kinarashah — COMMENTED** [review]
looks like we'd need to update to go1.17, from build: 
```
3466 | note: module requires Go 1.17
3467 | # k8s.io/ingress-nginx/test/e2e/settings/ocsp
3468 | test/e2e/settings/ocsp/ocsp.go:60:20: undefined: os.ReadFile
```

**Comment by brandond** [comment]
Not 1.17, I don't think upstream is using that yet. Docs say 1.16 which makes more sense.

https://pkg.go.dev/os#ReadFile

File: Dockerfile.dapper [hunk 1]
```diff
@@ -23,8 +23,8 @@ RUN apt-get update && \
     rm -f /bin/sh && ln -s /bin/bash /bin/sh
 ENV GOLANG_ARCH_amd64=amd64 GOLANG_ARCH_arm=armv6l GOLANG_ARCH_arm64=arm64 GOLANG_ARCH=GOLANG_ARCH_${ARCH} \
     GOPATH=/go PATH=/go/bin:/usr/local/go/bin:${PATH} SHELL=/bin/bash
-RUN wget -O - https://dl.google.com/go/go1.14.1.linux-${!GOLANG_ARCH}.tar.gz | tar -xzf - -C /usr/local && \
-    go get github.com/rancher/trash && go get golang.org/x/lint/golint && go get -u github.com/jteeuwen/go-bindata/...
+RUN wget -O - https://golang.org/dl/go1.16.6.linux-${!GOLANG_ARCH}.tar.gz | tar -xzf - -C /usr/local && \
```
Comment (brandond) [inline Dockerfile.dapper:26]: Why not
```suggestion
RUN wget -O - https://golang.org/dl/go1.16.8.linux-${!GOLANG_ARCH}.tar.gz | tar -xzf - -C /usr/local && \
```

**Comment by brandond** [comment]
Looks like we can drop 45d77db7564b01503e9f83f3380d2fa9dc87ac8d also.

**Comment by galal-hussein** [comment]
> Looks like we can drop [45d77db](https://github.com/rancher/ingress-nginx/commit/45d77db7564b01503e9f83f3380d2fa9dc87ac8d) also.

yup fixing that now

**Comment by brandond** [comment]
rather than reverting it I'd probably recommend just dropping it from the commit list when doing `git rebase`.

**Comment by galal-hussein** [comment]
> rather than reverting it I'd probably recommend just dropping it from the commit list when doing `git rebase -i`.

okidoke

**Review by kinarashah — APPROVED** [review]

**Pull request merged**